x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/InternalRealms.java

```diff
@@ -52,15 +54,22 @@
      */
     private static final Set<String> XPACK_TYPES = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
             NativeRealmSettings.TYPE, FileRealmSettings.TYPE, LdapRealmSettings.AD_TYPE, LdapRealmSettings.LDAP_TYPE, PkiRealmSettings.TYPE,
-            SamlRealmSettings.TYPE
+            SamlRealmSettings.TYPE, KerberosRealmSettings.TYPE
     )));
 
+    /**
+     * The list of all realm types, which are those that have extensive interaction
+     * with third party sources
+     */
+    private static final Set<String> TYPES_DEPEND_ON_THIRD_PARTY_SOURCES =
```

Comment:
I'd just remove this and change standard types to explicitly list the types. `Collections.unmodifiableSet(NativeRealmSettings.TYPE, FileRealmSettings.TYPE, LdapRealmSettings.AD_TYPE, LdapRealmSettings.LDAP_TYPE, PkiRealmSettings.TYPE);`